### PR TITLE
fungalize stages for zombies

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Content "fungalize stages for zombies"

#### Purpose of change

Realism

#### Describe the solution

Introduces two new stages of fungalization for base zombies, each taking one day to complete:

1. dusted zombies, they are covered with fungal spores and have begun to change, but for all intents and purposes are still zombies. These have a new faction that creates instant friendliness with fungus without instant full fungal brain control (don't immediately try to murder regular zombies)

2. fungal stalk zombies, they have a stalk of fungus coming out of their heads and have been taken over, but they aren't fully functional as fungal agents yet. They have very little movement and they can't spread the fungus yet.

Base zombies now fungalize into dusted zombies and then turn into the fungal stalk after a day.

#### Describe alternatives you've considered

Give these the ability to dust the PC or an NPC if they are grabbed, but I don't think we have a way to do that yet. 
Make dusted zombies only neutral instead of friendly, but couldn't make that work in testing. 
Use copy-from for dusted, but it created fungalize loops as they fungalized into themselves over and over.
Give fungal zombies grab so they can use their bite attacks. Good bugfix but out of scope for this PR IMO.

#### Testing

Monsters spawn as expected and behave as expected. 

Three zombies vs. three fungaloids:
Screenshot 2024-10-06 at 9 04 24 PM

Three zombies vs. three fungaloid shamblers:
Screenshot 2024-10-06 at 8 53 24 PM

#### Additional context

Inspired by the new fungal hulk, which instantly transforms from full zombie to full fungal. This doesn't solve that yet, but it is the first step.

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
Not that this isn't a good move, but this may cause a couple of issues I think are worth pointing out:

If you play as Mycus, your job is mostly to fungalize zombies to create minions for yourself as you conquer more territory, and this plays a lot like the cleric of Beogh from Dungeon Crawl Stone Soup. If this process now takes days, that becomes pointless or impossible, and you will just want to kill all the zombies yourself like normal.

Similarly, when zombie and fungal groups clash, the mycus is basically incapable of fighting back until it has fungalized zombies to work with. Currently the mycus will almost always win as the zombies have no means of fighting the infection, but now the zombies will (except against bosses) always wipe out the fungals, only to fungalize afterwards.

Workarounds are out of scope here, but imo bear mentioning. Fixes could include beefing up fungaloid shamblers, adding pre-turned zombies to fungal enemy spawns, and maybe giving Mycus mutants an expensive insta-turn option, the ability to get fungalized zombies to follow them over longer distances, summonable shamblers, or some other way to build an army.

**Comment by LyleSY** [comment]
Good call on the game balance issues. This is for sure a nerf on fungal power and growth, especially for day one before zombies start to ramp up hard. In my quick playtesting fungus is still dominant when numbers are even, but it takes a while.

**Comment by fairyarmadillo** [comment]
> Good call on the game balance issues. This is for sure a nerf on fungal power and growth, especially for day one before zombies start to ramp up hard. In my quick playtesting fungus is still dominant when numbers are even, but it takes a while.

A good fix for Mycus, and maybe for some fungal enemies, might be to give them the ability to instantly fungalize and revive an unpulped zombie corpse.

Narratively speaking (and do check with a dev here), I believe the Mycus is supposed to be a problem that the Blob can't rid itself of without expending considerable and concerted effort, but one that is generally beneath its notice. So having the conflict swing too hard in either side's favor is probably not wanted.

File: data/json/monsters/fungus_zombie.json [hunk 1]
```diff
@@ -1,4 +1,71 @@
...
+    "name": { "str": "fungal stalk zombie" },
+    "description": "This writhing corpse has grown a short stalk of gray fungus bursting from the skull.",
+    "copy-from": "mon_zombie_fungus",
+    "diff": 0,
+    "hp": 80,
+    "speed": 20,
+    "bleed_rate": 50,
+    "special_attacks": [ { "type": "bite", "cooldown": 5 } ],
+    "upgrades": { "age_grow": 1, "into": "mon_zombie_fungus" },
+    "flags": [ "SEES", "STUMBLES", "WARM", "BASHES", "GROUP_BASH", "POISON", "NO_BREATHE", "PUSH_MON", "FILTHY" ]
```
Comment (Night-Pryanik) [inline data/json/monsters/fungus_zombie.json:67]: Is there a reason that it's no longer able to hear?

Comment (LyleSY) [reply data/json/monsters/fungus_zombie.json:67]: I copied that from the final "mon_zombie_fungus" version. There my guess is that the ear canal is stuffed full and doesn't work anymore. I don't think that makes as much sense here. I'm restoring hearing for this step.

**Pull request merged**